packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx

```diff
@@ -298,7 +300,10 @@ const FilterConfiguration: FC<Props> = ({
                             selectedField && (
                                 <Group spacing="xs">
                                     <FieldIcon item={selectedField} />
-                                    <FieldLabel item={selectedField} />
+                                    <FieldLabel
```

Comment:
This component is not supposed to know about filters. 
We also want to keep the field name in edit mode so editors don't lose track of what field they are filtering.
I think we could rollback all the changes and have:
```
                                    {originalFilterRule?.label && !isEditMode ? (
                                        <Text span fw={500}>originalFilterRule.label</Text>
                                    ) : (
                                        <FieldLabel item={selectedField}/>
                                    )}
```